// naive-ui@2.38.2/es/themes/light.mjs
// The file is for internal usage, do not export it, since all the components
// have default light theme.
import { commonLight } from "../_styles/common/index.mjs";
import { alertLight } from "../alert/styles/index.mjs";
import { anchorLight } from "../anchor/styles/index.mjs";
import { autoCompleteLight } from "../auto-complete/styles/index.mjs";
import { avatarLight } from "../avatar/styles/index.mjs";
import { avatarGroupLight } from "../avatar-group/styles/index.mjs";
import { backTopLight } from "../back-top/styles/index.mjs";
import { badgeLight } from "../badge/styles/index.mjs";
import { breadcrumbLight } from "../breadcrumb/styles/index.mjs";
import { buttonLight } from "../button/styles/index.mjs";
import { buttonGroupLight } from "../button-group/styles/index.mjs";
import { calendarLight } from "../calendar/styles/index.mjs";
import { cardLight } from "../card/styles/index.mjs";
import { carouselLight } from "../carousel/styles/index.mjs";
import { cascaderLight } from "../cascader/styles/index.mjs";
import { checkboxLight } from "../checkbox/styles/index.mjs";
import { codeLight } from "../code/styles/index.mjs";
import { collapseLight } from "../collapse/styles/index.mjs";
import { collapseTransitionLight } from "../collapse-transition/styles/index.mjs";
import { colorPickerLight } from "../color-picker/styles/index.mjs";
import { dataTableLight } from "../data-table/styles/index.mjs";
import { datePickerLight } from "../date-picker/styles/index.mjs";
import { descriptionsLight } from "../descriptions/styles/index.mjs";
import { dialogLight } from "../dialog/styles/index.mjs";
import { dividerLight } from "../divider/styles/index.mjs";
import { drawerLight } from "../drawer/styles/index.mjs";
import { dropdownLight } from "../dropdown/styles/index.mjs";
import { dynamicInputLight } from "../dynamic-input/styles/index.mjs";
import { dynamicTagsLight } from "../dynamic-tags/styles/index.mjs";
import { elementLight } from "../element/styles/index.mjs";
import { ellipsisLight } from "../ellipsis/styles/index.mjs";
import { emptyLight } from "../empty/styles/index.mjs";
import { equationLight } from "../equation/styles/index.mjs";
import { formLight } from "../form/styles/index.mjs";
import { floatButtonLight } from "../float-button/styles/index.mjs";
import { gradientTextLight } from "../gradient-text/styles/index.mjs";
import { iconLight } from "../icon/styles/index.mjs";
import { iconWrapperLight } from "../icon-wrapper/styles/index.mjs";
import { imageLight } from "../image/styles/index.mjs";
import { inputLight } from "../input/styles/index.mjs";
import { inputNumberLight } from "../input-number/styles/index.mjs";
import { layoutLight } from "../layout/styles/index.mjs";
import { legacyTransferLight } from "../legacy-transfer/styles/index.mjs";
import { listLight } from "../list/styles/index.mjs";
import { loadingBarLight } from "../loading-bar/styles/index.mjs";
import { logLight } from "../log/styles/index.mjs";
import { menuLight } from "../menu/styles/index.mjs";
import { mentionLight } from "../mention/styles/index.mjs";
import { messageLight } from "../message/styles/index.mjs";
import { modalLight } from "../modal/styles/index.mjs";
import { notificationLight } from "../notification/styles/index.mjs";
import { pageHeaderLight } from "../page-header/styles/index.mjs";
import { paginationLight } from "../pagination/styles/index.mjs";
import { popconfirmLight } from "../popconfirm/styles/index.mjs";
import { popoverLight } from "../popover/styles/index.mjs";
import { popselectLight } from "../popselect/styles/index.mjs";
import { progressLight } from "../progress/styles/index.mjs";
import { qrcodeLight } from "../qr-code/styles/index.mjs";
import { radioLight } from "../radio/styles/index.mjs";
import { rateLight } from "../rate/styles/index.mjs";
import { resultLight } from "../result/styles/index.mjs";
import { rowLight } from "../legacy-grid/styles/index.mjs";
import { scrollbarLight } from "../_internal/scrollbar/styles/index.mjs";
import { selectLight } from "../select/styles/index.mjs";
import { skeletonLight } from "../skeleton/styles/index.mjs";
import { sliderLight } from "../slider/styles/index.mjs";
import { spaceLight } from "../space/styles/index.mjs";
import { spinLight } from "../spin/styles/index.mjs";
import { statisticLight } from "../statistic/styles/index.mjs";
import { stepsLight } from "../steps/styles/index.mjs";
import { switchLight } from "../switch/styles/index.mjs";
import { tableLight } from "../table/styles/index.mjs";
import { tabsLight } from "../tabs/styles/index.mjs";
import { tagLight } from "../tag/styles/index.mjs";
import { thingLight } from "../thing/styles/index.mjs";
import { timePickerLight } from "../time-picker/styles/index.mjs";
import { timelineLight } from "../timeline/styles/index.mjs";
import { tooltipLight } from "../tooltip/styles/index.mjs";
import { transferLight } from "../transfer/styles/index.mjs";
import { typographyLight } from "../typography/styles/index.mjs";
import { treeLight } from "../tree/styles/index.mjs";
import { treeSelectLight } from "../tree-select/styles/index.mjs";
import { uploadLight } from "../upload/styles/index.mjs";
import { watermarkLight } from "../watermark/styles/index.mjs";
import { splitLight } from "../split/styles/index.mjs";
import { flexLight } from "../flex/styles/index.mjs";
import { floatButtonGroupLight } from "../float-button-group/styles/index.mjs";
export const lightTheme = {
  name: 'light',
  common: commonLight,
  Alert: alertLight,
  Anchor: anchorLight,
  AutoComplete: autoCompleteLight,
  Avatar: avatarLight,
  AvatarGroup: avatarGroupLight,
  BackTop: backTopLight,
  Badge: badgeLight,
  Breadcrumb: breadcrumbLight,
  Button: buttonLight,
  ButtonGroup: buttonGroupLight,
  Calendar: calendarLight,
  Card: cardLight,
  Carousel: carouselLight,
  Cascader: cascaderLight,
  Checkbox: checkboxLight,
  Code: codeLight,
  Collapse: collapseLight,
  CollapseTransition: collapseTransitionLight,
  ColorPicker: colorPickerLight,
  DataTable: dataTableLight,
  DatePicker: datePickerLight,
  Descriptions: descriptionsLight,
  Dialog: dialogLight,
  Divider: dividerLight,
  Drawer: drawerLight,
  Dropdown: dropdownLight,
  DynamicInput: dynamicInputLight,
  DynamicTags: dynamicTagsLight,
  Element: elementLight,
  Empty: emptyLight,
  Equation: equationLight,
  Ellipsis: ellipsisLight,
  Flex: flexLight,
  Form: formLight,
  GradientText: gradientTextLight,
  Icon: iconLight,
  IconWrapper: iconWrapperLight,
  Image: imageLight,
  Input: inputLight,
  InputNumber: inputNumberLight,
  Layout: layoutLight,
  LegacyTransfer: legacyTransferLight,
  List: listLight,
  LoadingBar: loadingBarLight,
  Log: logLight,
  Menu: menuLight,
  Mention: mentionLight,
  Message: messageLight,
  Modal: modalLight,
  Notification: notificationLight,
  PageHeader: pageHeaderLight,
  Pagination: paginationLight,
  Popconfirm: popconfirmLight,
  Popover: popoverLight,
  Popselect: popselectLight,
  Progress: progressLight,
  QrCode: qrcodeLight,
  Radio: radioLight,
  Rate: rateLight,
  Row: rowLight,
  Result: resultLight,
  Scrollbar: scrollbarLight,
  Skeleton: skeletonLight,
  Select: selectLight,
  Slider: sliderLight,
  Space: spaceLight,
  Spin: spinLight,
  Statistic: statisticLight,
  Steps: stepsLight,
  Switch: switchLight,
  Table: tableLight,
  Tabs: tabsLight,
  Tag: tagLight,
  Thing: thingLight,
  TimePicker: timePickerLight,
  Timeline: timelineLight,
  Tooltip: tooltipLight,
  Transfer: transferLight,
  Tree: treeLight,
  TreeSelect: treeSelectLight,
  Typography: typographyLight,
  Upload: uploadLight,
  Watermark: watermarkLight,
  Split: splitLight,
  FloatButton: floatButtonLight,
  FloatButtonGroup: floatButtonGroupLight
};
