// naive-ui/es/themes/dark.mjs
import { commonDark } from "../_styles/common/index.mjs";
import { alertDark } from "../alert/styles/index.mjs";
import { anchorDark } from "../anchor/styles/index.mjs";
import { autoCompleteDark } from "../auto-complete/styles/index.mjs";
import { avatarDark } from "../avatar/styles/index.mjs";
import { avatarGroupDark } from "../avatar-group/styles/index.mjs";
import { backTopDark } from "../back-top/styles/index.mjs";
import { badgeDark } from "../badge/styles/index.mjs";
import { breadcrumbDark } from "../breadcrumb/styles/index.mjs";
import { buttonDark } from "../button/styles/index.mjs";
import { buttonGroupDark } from "../button-group/styles/index.mjs";
import { calendarDark } from "../calendar/styles/index.mjs";
import { cardDark } from "../card/styles/index.mjs";
import { carouselDark } from "../carousel/styles/index.mjs";
import { cascaderDark } from "../cascader/styles/index.mjs";
import { checkboxDark } from "../checkbox/styles/index.mjs";
import { codeDark } from "../code/styles/index.mjs";
import { collapseDark } from "../collapse/styles/index.mjs";
import { collapseTransitionDark } from "../collapse-transition/styles/index.mjs";
import { colorPickerDark } from "../color-picker/styles/index.mjs";
import { dataTableDark } from "../data-table/styles/index.mjs";
import { datePickerDark } from "../date-picker/styles/index.mjs";
import { descriptionsDark } from "../descriptions/styles/index.mjs";
import { dialogDark } from "../dialog/styles/index.mjs";
import { dividerDark } from "../divider/styles/index.mjs";
import { drawerDark } from "../drawer/styles/index.mjs";
import { dropdownDark } from "../dropdown/styles/index.mjs";
import { dynamicInputDark } from "../dynamic-input/styles/index.mjs";
import { dynamicTagsDark } from "../dynamic-tags/styles/index.mjs";
import { elementDark } from "../element/styles/index.mjs";
import { ellipsisDark } from "../ellipsis/styles/index.mjs";
import { emptyDark } from "../empty/styles/index.mjs";
import { equationDark } from "../equation/styles/index.mjs";
import { formDark } from "../form/styles/index.mjs";
import { floatButtonDark } from "../float-button/styles/index.mjs";
import { gradientTextDark } from "../gradient-text/styles/index.mjs";
import { iconDark } from "../icon/styles/index.mjs";
import { iconWrapperDark } from "../icon-wrapper/styles/index.mjs";
import { imageDark } from "../image/styles/index.mjs";
import { inputDark } from "../input/styles/index.mjs";
import { inputNumberDark } from "../input-number/styles/index.mjs";
import { layoutDark } from "../layout/styles/index.mjs";
import { legacyTransferDark } from "../legacy-transfer/styles/index.mjs";
import { listDark } from "../list/styles/index.mjs";
import { loadingBarDark } from "../loading-bar/styles/index.mjs";
import { logDark } from "../log/styles/index.mjs";
import { menuDark } from "../menu/styles/index.mjs";
import { mentionDark } from "../mention/styles/index.mjs";
import { messageDark } from "../message/styles/index.mjs";
import { modalDark } from "../modal/styles/index.mjs";
import { notificationDark } from "../notification/styles/index.mjs";
import { pageHeaderDark } from "../page-header/styles/index.mjs";
import { paginationDark } from "../pagination/styles/index.mjs";
import { popconfirmDark } from "../popconfirm/styles/index.mjs";
import { popoverDark } from "../popover/styles/index.mjs";
import { popselectDark } from "../popselect/styles/index.mjs";
import { progressDark } from "../progress/styles/index.mjs";
import { qrcodeDark } from "../qr-code/styles/index.mjs";
import { radioDark } from "../radio/styles/index.mjs";
import { rateDark } from "../rate/styles/index.mjs";
import { resultDark } from "../result/styles/index.mjs";
import { rowDark } from "../legacy-grid/styles/index.mjs";
import { scrollbarDark } from "../_internal/scrollbar/styles/index.mjs";
import { selectDark } from "../select/styles/index.mjs";
import { skeletonDark } from "../skeleton/styles/index.mjs";
import { sliderDark } from "../slider/styles/index.mjs";
import { spaceDark } from "../space/styles/index.mjs";
import { spinDark } from "../spin/styles/index.mjs";
import { statisticDark } from "../statistic/styles/index.mjs";
import { stepsDark } from "../steps/styles/index.mjs";
import { switchDark } from "../switch/styles/index.mjs";
import { tableDark } from "../table/styles/index.mjs";
import { tabsDark } from "../tabs/styles/index.mjs";
import { tagDark } from "../tag/styles/index.mjs";
import { thingDark } from "../thing/styles/index.mjs";
import { timePickerDark } from "../time-picker/styles/index.mjs";
import { timelineDark } from "../timeline/styles/index.mjs";
import { tooltipDark } from "../tooltip/styles/index.mjs";
import { transferDark } from "../transfer/styles/index.mjs";
import { treeSelectDark } from "../tree-select/styles/index.mjs";
import { typographyDark } from "../typography/styles/index.mjs";
import { treeDark } from "../tree/styles/index.mjs";
import { uploadDark } from "../upload/styles/index.mjs";
import { watermarkDark } from "../watermark/styles/index.mjs";
import { splitDark } from "../split/styles/index.mjs";
import { flexDark } from "../styles.mjs";
import { floatButtonGroupDark } from "../float-button-group/styles/index.mjs";
export const darkTheme = {
  name: 'dark',
  common: commonDark,
  Alert: alertDark,
  Anchor: anchorDark,
  AutoComplete: autoCompleteDark,
  Avatar: avatarDark,
  AvatarGroup: avatarGroupDark,
  BackTop: backTopDark,
  Badge: badgeDark,
  Breadcrumb: breadcrumbDark,
  Button: buttonDark,
  ButtonGroup: buttonGroupDark,
  Calendar: calendarDark,
  Card: cardDark,
  Carousel: carouselDark,
  Cascader: cascaderDark,
  Checkbox: checkboxDark,
  Code: codeDark,
  Collapse: collapseDark,
  CollapseTransition: collapseTransitionDark,
  ColorPicker: colorPickerDark,
  DataTable: dataTableDark,
  DatePicker: datePickerDark,
  Descriptions: descriptionsDark,
  Dialog: dialogDark,
  Divider: dividerDark,
  Drawer: drawerDark,
  Dropdown: dropdownDark,
  DynamicInput: dynamicInputDark,
  DynamicTags: dynamicTagsDark,
  Element: elementDark,
  Empty: emptyDark,
  Ellipsis: ellipsisDark,
  Equation: equationDark,
  Flex: flexDark,
  Form: formDark,
  GradientText: gradientTextDark,
  Icon: iconDark,
  IconWrapper: iconWrapperDark,
  Image: imageDark,
  Input: inputDark,
  InputNumber: inputNumberDark,
  LegacyTransfer: legacyTransferDark,
  Layout: layoutDark,
  List: listDark,
  LoadingBar: loadingBarDark,
  Log: logDark,
  Menu: menuDark,
  Mention: mentionDark,
  Message: messageDark,
  Modal: modalDark,
  Notification: notificationDark,
  PageHeader: pageHeaderDark,
  Pagination: paginationDark,
  Popconfirm: popconfirmDark,
  Popover: popoverDark,
  Popselect: popselectDark,
  Progress: progressDark,
  QrCode: qrcodeDark,
  Radio: radioDark,
  Rate: rateDark,
  Result: resultDark,
  Row: rowDark,
  Scrollbar: scrollbarDark,
  Select: selectDark,
  Skeleton: skeletonDark,
  Slider: sliderDark,
  Space: spaceDark,
  Spin: spinDark,
  Statistic: statisticDark,
  Steps: stepsDark,
  Switch: switchDark,
  Table: tableDark,
  Tabs: tabsDark,
  Tag: tagDark,
  Thing: thingDark,
  TimePicker: timePickerDark,
  Timeline: timelineDark,
  Tooltip: tooltipDark,
  Transfer: transferDark,
  Tree: treeDark,
  TreeSelect: treeSelectDark,
  Typography: typographyDark,
  Upload: uploadDark,
  Watermark: watermarkDark,
  Split: splitDark,
  FloatButton: floatButtonDark,
  FloatButtonGroup: floatButtonGroupDark
};
